### PR TITLE
fix: add performance optimisations

### PR DESCRIPTION
**Pull request by apurv-wednesday**
* update the rendering logic
* add colour constants
* update translations
* fix jest config

## Summary by CodeRabbit

- **New Features**
  - Enhanced repository browsing experience with updated URL parameter handling and improved visual styling in repository components.
  - Added dynamic color changes in error states based on theme settings to improve visual feedback.
- **Enhancements**
  - Improved internationalization with new translations and updated message IDs for better localization support.
- **Bug Fixes**
  - Fixed issues with incorrect or outdated color codes in UI components.
- **Refactor**
  - Optimized configuration settings for better project structure and maintainability.
- **Documentation**
  - Updated documentation to reflect new features and enhancements.

### CODE REVIEW

File: jest.config.ts [hunk 1]
```diff
@@ -7,14 +7,14 @@ const jestConfig: Config = {
...
   collectCoverageFrom: [
-    "./common/**/*.{js,jsx,ts,tsx}",
-    "./containers/**/*.{js,jsx,ts,tsx}",
-    "./features/**/*.{js,jsx,ts,tsx}",
-    "./pages/**/*.{js,jsx,ts,tsx}",
-    "./store/**/*.{js,jsx,ts,tsx}",
-    "./styles/**/*.{js,jsx,ts,tsx}",
-    "./themes/**/*.{js,jsx,ts,tsx}",
-    "./utils/**/*.{js,jsx,ts,tsx}",
+    "src/common/**/*.{js,jsx,ts,tsx}",
```
Comment (alichherawalla) [inline jest.config.ts:10]: this can just become "src/**/*.{js,jsx,ts,tsx}"

File: src/containers/Repos/index.tsx [hunk 2]
```diff
@@ -8,68 +8,84 @@ import { useFetchRecommendationQuery } from "@features/repos/api/getRecommendati
...
   useEffect(() => {
     if (router.isReady) {
-      setRepoName(router.query?.search as string);
-      setPage(router.query?.page as unknown as number);
+      const searchParam = router.query?.search as string;
+      setRepoName(searchParam || "");
+      setPage(Number(router.query?.page) || 1);
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (alichherawalla) [inline src/containers/Repos/index.tsx:64]: can we remove the eslint-disable-next-line? doesn't seem like we need it neccessarily here

File: src/containers/Repos/index.tsx [hunk 3]
```diff
@@ -8,68 +8,84 @@ import { useFetchRecommendationQuery } from "@features/repos/api/getRecommendati
...
   }, [router.isReady]);
 
   useEffect(() => {
     if (!isEmpty(repoName)) {
-      router.push(`/?search=${repoName}&page=${page}`);
+      updateUrlParams(repoName, page);
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (alichherawalla) [inline src/containers/Repos/index.tsx:71]: please check if we can remove

File: src/features/repos/components/ErrorState/tests/__snapshots__/index.test.tsx.snap [hunk 4]
```diff
@@ -15,8 +15,8 @@ exports[`<ErrorState /> should render and match the snapshot 1`] = `
   margin: 20px 0;
   padding: 1rem;
   max-width: px;
-  color: grey;
-  color: grey;
+  color: [object Object];
```
Comment (alichherawalla) [inline src/features/repos/components/ErrorState/tests/__snapshots__/index.test.tsx.snap:18]: this appears broken

File: src/features/repos/components/ErrorState/tests/__snapshots__/index.test.tsx.snap [hunk 5]
```diff
@@ -41,7 +41,7 @@ exports[`<ErrorState /> should render and match the snapshot 1`] = `
   <div>
     <div
       class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root emotion-0"
-      color="grey"
+      color="[object Object]"
```
Comment (alichherawalla) [inline src/features/repos/components/ErrorState/tests/__snapshots__/index.test.tsx.snap:44]: this too

File: src/containers/Repos/index.tsx (hunk 2 above)
Comment (apurv-wednesday) [reply src/containers/Repos/index.tsx]: removed

File: src/containers/Repos/index.tsx (hunk 3 above)
Comment (apurv-wednesday) [reply src/containers/Repos/index.tsx]: removed

File: jest.config.ts (hunk 1 above)
Comment (apurv-wednesday) [reply jest.config.ts]: Intentionally left like that for better flexibility in coverage/exclusion.
Replaced with src/ for better readability.

File: src/features/repos/components/ErrorState/tests/__snapshots__/index.test.tsx.snap (hunk 4 above)
Comment (apurv-wednesday) [reply src/features/repos/components/ErrorState/tests/__snapshots__/index.test.tsx.snap]: fixed

File: jest.config.ts (hunk 1 above)
Comment (apurv-wednesday) [reply jest.config.ts]: fixed

**Review by alichherawalla — APPROVED** [review]

**Pull request merged**